pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -72,6 +77,29 @@ def _updateDataRect(self):
             self.containsNonfinite = not (all_x_are_finite and all_y_are_finite) # This always yields a definite True/False answer
         self._dataRect = QtCore.QRectF( QtCore.QPointF(xmin,ymin), QtCore.QPointF(xmax,ymax) )
         
+    def append(self, x, y):
```

Comment:
if we _might_ be concerned regarding the performance here, I would consider adding the following bit as the first line of code in this function.

```python

def append(self, x, y):
    """
    docstring
    """
    profiler = pg.debug.Profiler()  # add this line
    old_length = len(self.y)
    .....
```

Having a profiler in a particular function is usually a decent indication that we're concerned about the performance here, and makes it simpler to modify the line so it actually spits output on the console if we decide to investigate fruther.